packages/leva/src/types/public.ts

```diff
@@ -204,7 +224,7 @@ export type Leaves<IncludeTransient extends boolean, T, P extends string | numbe
   ? T extends object
     ? 3
     : 4
-  : T extends { onChange: any } // if an input has the onChange property then it's transient and isn't returned
+  : T extends TransientOnChangeSchemaItemOptions // if an input has the onChange property then it's transient and isn't returned
```

Comment:
Wouldn't this be simpler?

```ts
  : T extends { onChange: any; transient?: true } // if an input has the onChange property then it's transient and isn't returned
```

...without needing `OnChangeSchemaItemOptions, NoOnChangeSchemaItemOptions, NonTransientOnChangeSchemaItemOptions, TransientOnChangeSchemaItemOptions`?

Unless I'm missing something?

PS: at the moment we don't check if `onChange` is a function, only if it's non null, so the Typescript check should reflect this I think.